src/sqlfluff/rules/L026.py

```diff
@@ -28,6 +28,17 @@ class Rule_L026(Rule_L025):
 
     """
 
+    @staticmethod
+    def _is_bad_tbl_ref(table_aliases, parent_select, tbl_ref):
+        if tbl_ref and tbl_ref[0] not in [a.ref_str for a in table_aliases]:
+            # Last check, this *might* be a correlated subquery reference.
+            if parent_select:
+                parent_aliases, _ = get_aliases_from_select(parent_select)
+                if parent_aliases and tbl_ref[0] in [a[0] for a in parent_aliases]:
```

Comment:
could do
```suggestion
                return not (parent_aliases and tbl_ref[0] in [a[0] for a in parent_aliases])
```

Reply:
Makes sense, but I think I'd like to keep the existing structure -- now that I have added more comments as you suggested, I find that the various `return` statements are good places to provide context on the significance and context of reaching various lines of code. I'll push up this changes in a few minutes.